### PR TITLE
ci: don't test the master FIPS provider against a 3.0 build

### LINKED ISSUE #19601: Across version provider testing is failing
The test that checks that the 3.0 and master FIPS providers interoperate is [broken](https://github.com/paulidale/openssl/actions/runs/3388207103/jobs/5629908581).

In master and 3.1, triple DES has been removed from the FIPS provider because of an end of 2023 sunset.  Refer [SP 800-131Ar2](https://nvlpubs.nist.gov/nistpubs/SpecialPublications/NIST.SP.800-131Ar2.pdf).  The unit tests in 3.0 haven't the slightest clue that this is happening and fail because the triple DES algorithms cannot be fetched and tested.

Worse is that PKCS **⌗**1 version 1.5 padding is also being sunsetted at the end of 2023 and it's upcoming removal from the FIPS provider in master and 3.1 will cause a lot more errors to happen in the 3.0 cross version testing.

The underlying problem of algorithms going missing and been modified needs to be addressed somehow.

### PR DESCRIPTION
**Pull request by paulidale**
The FIPS provider in master & 3.1 cannot pass the tests present in a 3.0 build.

This is a result of the removal of triple DES from the provider. The future removal of PKCS #1 v1.5 padding will exacerbate the failures further.

Fixes #19601

- [ ] documentation is added or updated
- [x] tests are added or updated

### CODE REVIEW

**Comment by slontis** [comment]
To do this properly, the 3.0 tests would need to be modified to get the provider version, or query if an algorithm is available..

**Comment by paulidale** [comment]
Modifying the 3.0 tests wouldn't qualify as a bug fix IMO.

**Comment by slontis** [comment]
We decided to do these tests for a reason, Disabling them after a such a short time seems like a bad idea.

**Comment by slontis** [comment]
Might need a OTC hold?

**Comment by levitte** [comment]
This raises legitimate questions regarding test designs.

It would probably be smart to have a set of tests that are meant to exercise the FIPS provider, and apart from that would rely on the public API from OpenSSL 3.0.  That would form a package of tests that belong with the FIPS provider version rather than with the OpenSSL version, but that can still be built against any OpenSSL version that the FIPS provider is compatible with.

To put it simply, to test the master branch FIPS provider against OpenSSL 3.0, it might be more productive to use the master branch test suite, but linked with OpenSSL 3.0 libraries, or if that's enough, use the master branch evp_test test vectors with an OpenSSL 3.0 evp_test.

**Review by t8m — APPROVED** [review]
Agreed with urgent. If we want some better solution, we can do it later.

**Comment by t8m** [comment]
Alternative quick-fix solution, perhaps better than dropping the ci build altogether, would be to exclude the testcases that currently fail.

Longer term solution would be to ship a patch file against the openssl-3.0 and apply it before running the tests.

**Comment by mattcaswell** [comment]
> We decided to do these tests for a reason, Disabling them after a such a short time seems like a bad idea.

Agreed. But I also agree that we need to fix the CI asap.

We should probably include some work within the 140-3 project to fix the 3.0 tests to be able to cope with this in some way.

> Modifying the 3.0 tests wouldn't qualify as a bug fix IMO.

True, but we may want to consider an exception for this.

**Comment by paulidale** [comment]
The upcoming PKCS #1 v1.5 removals will cause a **lot more pain** than the TDES removal has.
Forward compatibility is reasonable, backward isn't in light of the upcoming transitions.

**Comment by slontis** [comment]
This needs a OTC discussion I think.

**Comment by paulidale** [comment]
The transitioned algorithms will be retained inside the 140-3 FIPS provider but flagged as not validated.  This will allow the tests to pass again.  Refer to #19631.
